### PR TITLE
fix street names

### PR DESCRIPTION
**Pull request by nnngrach**

### CODE REVIEW

File: native/src/routeSegmentResult.h [hunk 1]
```diff
@@ -161,6 +164,79 @@ struct RouteSegmentResult {
 		}
 		return res;
 	}
+
```
Comment (vshcherb) [inline native/src/routeSegmentResult.h:167]: 1. Too much code in H file
2. Inline is not needed

File: native/src/routeSegmentResult.h [hunk 2]
```diff
@@ -161,6 +164,79 @@ struct RouteSegmentResult {
...
+	}
+	
+	inline SHARED_PTR<RouteDataObject> getObjectWithShield(vector<SHARED_PTR<RouteSegmentResult>> list, int routeInd) {
+		SHARED_PTR<RouteDataObject> rdo;
+		bool isNextShieldFound = object->hasNameTagStartsWith("road_ref");
+		for (int ind = routeInd; ind < list.size() && !isNextShieldFound; ind++) {
+			if (list[ind]->turnType != nullptr) {
+				isNextShieldFound = true;
+			} else {
+				SHARED_PTR<RouteDataObject> obj = list[ind]->object;
```
Comment (vshcherb) [inline native/src/routeSegmentResult.h:230]: should be reference ```SHARED_PTR<RouteDataObject> ? ``` ?

File: native/src/routeSegmentResult.h [hunk 3]
```diff
@@ -161,6 +164,79 @@ struct RouteSegmentResult {
...
+	}
+	
+	inline string getRef(string lang, bool transliterate) {
+		return object->getRef(lang, transliterate, isForwardDirection());
+	}
+	
+	inline SHARED_PTR<RouteDataObject> getObjectWithShield(vector<SHARED_PTR<RouteSegmentResult>> list, int routeInd) {
+		SHARED_PTR<RouteDataObject> rdo;
+		bool isNextShieldFound = object->hasNameTagStartsWith("road_ref");
+		for (int ind = routeInd; ind < list.size() && !isNextShieldFound; ind++) {
```
Comment (vshcherb) [inline native/src/routeSegmentResult.h:226]: uint instead of int - check compilation warnings!!

File: native/src/routeSegmentResult.h [hunk 4]
```diff
@@ -69,6 +72,11 @@ struct RouteSegmentResult {
 	LatLon getStartPoint();
 	LatLon getEndPoint();
 	LatLon getPoint(int i);
+	
+	string getDestinationName(string lang, bool transliterate, vector<SHARED_PTR<RouteSegmentResult>> list, int routeInd);
+	string getStreetName(string lang, bool transliterate, vector<SHARED_PTR<RouteSegmentResult>> list, int routeInd);
+	string getRef(string lang, bool transliterate);
+	SHARED_PTR<RouteDataObject> getObjectWithShield(vector<SHARED_PTR<RouteSegmentResult>> list, uint routeInd);
```
Comment (alex-osm) [inline native/src/routeSegmentResult.h:79]: vector≤≥& ?

**Pull request merged**